### PR TITLE
*: make TestStores stable

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: Close https://github.com/tikv/pd/issues/7954

### What is changed and how does it work?

```commit-message
add a failpoint to make sure the offline store do not convert to tombstone
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: tests/integrations/mcs/scheduling/api_test.go [hunk 1]
```diff
@@ -629,13 +629,13 @@ func (suite *apiTestSuite) checkStores(cluster *tests.TestCluster) {
 			Version:   "2.0.0",
 		},
 	}
-	// prevent the offline store from changing to tombstone
-	tests.MustPutRegion(re, cluster, 3, 6, []byte("a"), []byte("b"))
```
Comment (okJiang) [inline tests/integrations/mcs/scheduling/api_test.go:633]: I found tikv6 has been buried, so I am not sure about this line works.


```
[2024/08/13 08:59:41.814 +00:00] [WARN] [cluster.go:1437] ["store has been Tombstone"] [store-id=6] [store-address=tikv6] [state=Offline] [physically-destroyed=false]
```

**Review by nolouch — APPROVED** [review]

File: tests/integrations/mcs/scheduling/api_test.go (hunk 1 above)
Comment (rleungx) [reply tests/integrations/mcs/scheduling/api_test.go:633]: I was wondering why line 633 doesn't take effect.

Comment (okJiang) [reply tests/integrations/mcs/scheduling/api_test.go:633]: I haven't delved into it, it looks like a long chain.

File: tests/integrations/mcs/scheduling/api_test.go [hunk 2]
```diff
@@ -629,13 +629,13 @@ func (suite *apiTestSuite) checkStores(cluster *tests.TestCluster) {
...
+
+	re.NoError(failpoint.Enable("github.com/tikv/pd/server/cluster/DoNotBuryStore", `return(true)`))
+	defer func() {
+		re.NoError(failpoint.Disable("github.com/tikv/pd/server/cluster/DoNotBuryStore"))
```
Comment (rleungx) [inline tests/integrations/mcs/scheduling/api_test.go:635]:
```suggestion
		re.NoError(failpoint.Disable("github.com/tikv/pd/server/cluster/doNotBuryStore"))
```

**Review by rleungx — APPROVED** [review]

**Pull request merged**